cmake/tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -2,78 +2,103 @@
 # See associated tribits/Copyright.txt file for copyright and license! #
 ########################################################################
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding HDF5.  These are used in case find_package(HDF5 ...) is not called
-# or does not find HDF5.  Also, these variables need to be non-null in order
-# to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
-
-set(REQUIRED_HEADERS hdf5.h)
-set(REQUIRED_LIBS_NAMES hdf5)
-
-if (HDF5_REQUIRE_FORTRAN)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_fortran)
-endif()
+set(HDF5_INTERNAL_IS_MODERN FALSE)
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} z)
-endif()
+if (Netcdf_ALLOW_MODERN)
+
+  set(minimum_modern_HDF5_version 1.13.2)
+  print_var(Netcdf_ALLOW_MODERN)
+  message("-- Using find_package(HDF5 ${minimum_modern_HDF5_version} CONFIG) ...")
+  find_package(HDF5  ${minimum_modern_HDF5_version}  CONFIG)
+  if (HDF5_FOUND)
+    message("-- Found HDF5_CONFIG=${HDF5_CONFIG}")
+    message("-- Generating Netcdf::all_libs and NetcdfConfig.cmake")
+    message("-- HDF5_EXPORT_LIBRARIES=${HDF5_EXPORT_LIBRARIES}")
+    tribits_extpkg_create_imported_all_libs_target_and_config_file(
+      HDF5
+      INNER_FIND_PACKAGE_NAME  HDF5
+      IMPORTED_TARGETS_FOR_ALL_LIBS   ${HDF5_EXPORT_LIBRARIES})
+    set(HDF5_INTERNAL_IS_MODERN TRUE)
+  endif()
 
-if (TPL_ENABLE_Netcdf)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_hl)
 endif()
 
-#
-# Second, search for HDF5 components (if allowed) using the standard
-# find_package(HDF5 ...).
-#
-tribits_tpl_allow_pre_find_package(HDF5  HDF5_ALLOW_PREFIND)
-if (HDF5_ALLOW_PREFIND)
+set(HDF5_FOUND_MODERN_CONFIG_FILE ${HDF5_INTERNAL_IS_MODERN} CACHE INTERNAL "True if HDF5 was found by the modern method")
 
-  message("-- Using find_package(HDF5 ...) ...")
+if (NOT TARGET HDF5::all_libs)
+
+  # First, set up the variables for the (backward-compatible) TriBITS way of
+  # finding HDF5.  These are used in case find_package(HDF5 ...) is not called
+  # or does not find HDF5.  Also, these variables need to be non-null in order
+  # to trigger the right behavior in the function
+  # tribits_tpl_find_include_dirs_and_libraries().
+
+  set(REQUIRED_HEADERS hdf5.h)
+  set(REQUIRED_LIBS_NAMES hdf5)
 
-  set(HDF5_COMPONENTS C)
   if (HDF5_REQUIRE_FORTRAN)
-    list(APPEND HDF5_COMPONENTS Fortran)
+    set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_fortran)
   endif()
 
   if (TPL_ENABLE_MPI)
-    set(HDF5_PREFER_PARALLEL TRUE)
+    set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} z)
```

Comment:
`libz` is required for serial builds also.  Is there some issue where it isn't specified correctly in the parallel build/config?